urlcdab7d8364_jrwalsh_CycTools_tgz-pJ8-edu_iastate_cyctools_MainJ8/src/edu/iastate/cyctools/CycDataBaseAccess.java

```diff
@@ -0,0 +1,687 @@
+				
+				for (String slotLabel : frame.getSlots().keySet()) {
+					Element slotElement = doc.createElement("Slot");
+					slotElement.setAttribute("Label", slotLabel);
+					
+					ArrayList<Object> slotValues = frame.getSlotValues(slotLabel); //Sometimes slot data is actually an arraylist, not a string
+					for (Object slotValueObject : slotValues) {
+						Element slotValueElement = doc.createElement("SlotValue");
+						
+						if (slotValueObject instanceof ArrayList) {
+							// Handle when the slot value is actually an array of values
+							ArrayList<String> valueArray = (ArrayList<String>)slotValueObject;
+							for (String value : valueArray) {
+								Element valueElement = doc.createElement("Value");
+								valueElement.appendChild(doc.createTextNode(value));
+								slotValueElement.appendChild(valueElement);
+							}
+						} else {
+							// Handle when slot value is a normal string value
+							String slotValue = (String) slotValueObject;
+							
+							Element valueElement = doc.createElement("Value");
+							valueElement.appendChild(doc.createTextNode(slotValue));
+							slotValueElement.appendChild(valueElement);
+							for (String annotationLabel : frame.getAllAnnotLabels(slotLabel, slotValue)) {
+								Element annotationElement = doc.createElement("Annotation");
+								annotationElement.setAttribute("Label", annotationLabel);
+								slotValueElement.appendChild(annotationElement);
+								
+								ArrayList<String> annotationValues = frame.getAnnotations(slotLabel, slotValue, annotationLabel);
+								for (String annotationValue : annotationValues) {
+									Element annotationValueElement = doc.createElement("Value");
+									annotationValueElement.appendChild(doc.createTextNode(annotationValue));
+									annotationElement.appendChild(annotationValueElement);
+								}
+							}
+						}
+						slotElement.appendChild(slotValueElement);
+					}
+					frameElement.appendChild(slotElement);
+				}
+				rootElement.appendChild(frameElement);
+			}
+	 
+			// write the content into xml file
+			TransformerFactory transformerFactory = TransformerFactory.newInstance();
+			Transformer transformer = transformerFactory.newTransformer();
+			DOMSource source = new DOMSource(doc);
+			StreamResult result = new StreamResult(new File(path));
+	 
+			// StreamResult result = new StreamResult(System.out);
+	 
+			transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+			transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+			transformer.transform(source, result);
+	 
+//			System.out.println("File saved!");
+			
+			return 0;
+	  	} catch (PtoolsErrorException e) {
+	  		e.printStackTrace();
+	  	} catch (ParserConfigurationException pce) {
+	  		pce.printStackTrace();
+	  	} catch (TransformerException tfe) {
+	  		tfe.printStackTrace();
+	  	}
+		return -1;
+	}
+	
+	public int printFramesToCSV(String path, String type) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.